python/ray/_private/async_compat.py

```diff
@@ -27,13 +28,20 @@ def try_install_uvloop():
         pass
 
 
-def is_async_func(func):
+def is_async_func(func) -> bool:
     """Return True if the function is an async or async generator method."""
     return inspect.iscoroutinefunction(func) or inspect.isasyncgenfunction(func)
 
 
-def sync_to_async(func):
-    """Convert a blocking function to async function"""
+@lru_cache(maxsize=2**10)
+def has_async_methods(cls: object) -> bool:
+    """Return True if the class has any async methods."""
+    return len(inspect.getmembers(cls, predicate=is_async_func)) > 0
+
+
+@lru_cache(maxsize=2**10)
+def sync_to_async(func) -> bool:
```

Comment:
The return type is not bool

Reply:
D'oh, good catch, I'll fix this tomorrow.